src/app/(proper_react)/redesign/(authenticated)/admin/feature-flags/ModifyInputField.tsx

```diff
@@ -0,0 +1,40 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+"use client";
+
+export const ModifyInputField = (props: {
+  id: string | undefined;
+  name: string | undefined;
+  defaultValue: string | undefined;
+}) => {
+  // @ts-ignore FIXME determine correct event type
+  const handleChange = void (async (event) => {
+    const id = event.target.id;
+    const name = event.target.name;
+    const value = event.target.value;
```

Comment:
This code looks similar to the previous file, except the previous file had a bunch of optional chaining `?.` stuffs for `event` and `target`:

```js
    const data = {
      name: event?.target?.name.value,
      …
```

Reply:
Yeah I think the better approach is going to be defining the type here and using a type guard to catch problems at run time, I'm trying to figure out if I can re-use the existing `FeatureFlag` type since it's really close.

Reply:
(`event.target` is basically a string-ified version of `FeatureFlag` with `.value` on each leaf node, I'm not sure if there's an elegant way to declare that for the typescript compiler)

Reply:
OK I declared in the interface that these fields can be null more consistently, and put really basic checking in place (I think we could do a more abstract type guard but I'd like to coordinate with front-end folks who know this stuff better)